app/src/main/java/zapsolutions/zap/walletManagement/WalletDetailsActivity.java

```diff
@@ -0,0 +1,208 @@
+            tvHost.setText(getWalletConfig().getHost());
+            TextView tvPortLabel = findViewById(R.id.portLabel);
+            tvPortLabel.setText(getResources().getString(R.string.port) + ":");
+            TextView tvPort = findViewById(R.id.port);
+            tvPort.setText(String.valueOf(getWalletConfig().getPort()));
+            TextView tvMacaroonLabel = findViewById(R.id.macaroonLabel);
+            tvMacaroonLabel.setText(getResources().getString(R.string.macaroon) + ":");
+            TextView tvMacaroon = findViewById(R.id.macaroon);
+            tvMacaroon.setText(getWalletConfig().getMacaroon());
+            TextView tvCertificateLabel = findViewById(R.id.certLabel);
+            tvCertificateLabel.setText(getResources().getString(R.string.certificate) + ":");
+            TextView tvCertificate = findViewById(R.id.cert);
+            tvCertificate.setText(getWalletConfig().getCert());
+
+            Button changeBtn = findViewById(R.id.buttonChangeConnection);
+            changeBtn.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Intent intent = new Intent(WalletDetailsActivity.this, ConnectRemoteNodeActivity.class);
+                    intent.putExtra(ManageWalletsActivity.WALLET_ID, mId);
+                    startActivity(intent);
+                }
+            });
+        }
+
+        Button switchBtn = findViewById(R.id.buttonActivate);
+        switchBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                PrefsUtil.edit().putString(PrefsUtil.CURRENT_WALLET_CONFIG, mId).commit();
+
+                // Do not ask for pin again...
+                TimeOutUtil.getInstance().restartTimer();
+
+                Wallet.getInstance().reset();
+
+                if (!getWalletConfig().isLocal()) {
+                    // ToDo: Stop local LND if we switch to a remote wallet (right now it stops, but then it crashes when switching back to Local LND
+                    openHome();
+                } else {
+                    openHome();
+                }
+            }
+        });
+
+        Button renameBtn = findViewById(R.id.buttonRename);
+        renameBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                showWalletNameInput();
+            }
+        });
+
+        Button deleteBtn = findViewById(R.id.buttonDelete);
+        deleteBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (PrefsUtil.getCurrentWalletConfig().equals(mId)) {
+                    Toast.makeText(WalletDetailsActivity.this, "You cannot delete the currently active wallet.", Toast.LENGTH_LONG).show();
+                } else {
+                    deleteWallet();
```

Comment:
I think we should show a confirmation dialog here. 
Just to make sure user not accidentally clicks on this button.

Reply:
I have added the dialog, feels better now!